bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/FeatureInstaller.java

```diff
@@ -389,65 +392,90 @@ private synchronized void installFeatures(FeaturesService featuresService, Set<S
             featuresService.installFeatures(addons,
                     EnumSet.of(FeaturesService.Option.Upgrade, FeaturesService.Option.NoFailOnFeatureNotFound));
             logger.debug("Installed '{}'", StringUtils.join(addons, ", "));
-            for (String addon : addons) {
-                if (isInstalled(featuresService, addon)) {
-                    postInstalledEvent(addon);
+            try {
+                Feature[] features = featuresService.listInstalledFeatures();
+                for (String addon : addons) {
+                    if (isInstalled(features, addon)) {
+                        postInstalledEvent(addon);
+                    }
                 }
+            } catch (Exception e) {
+                logger.error("Failed retrieving features: {}", e.getMessage());
```

Comment:
If you catch all `Exception` you are potentially also catching for example `NullPointerException`s and  they tend to miss a message, so you will be printing the message `null` which will make it troublesome to debug, in this case it is always better to log the complete stacktrace.